packages/core/test/assign.test.ts

```diff
@@ -77,7 +77,7 @@ const counterMachine = Machine<CounterContext>({
         SET_MAYBE: [
           {
             actions: [
-              assign({
+              assign<CounterContext>({
```

Comment:
Trying to figure out why this won't work without the generic. It did before.

Reply:
I don't understand this either - I've tried to understand how TS infers this kind of information behavior and I've failed. There was some other case before then inference in a similar position has failed but I couldn't make sense out of it. Would be great to get help from some TS expert in explaining such things to us :<

Reply:
We'll defer this for now, there's already a TypeScript note in the docs re: this.